### PR TITLE
Improve histogram performance

### PR DESCRIPTION
**Pull request by bryanlb**
Swaps the internal implementation in the histogram builder to use a faster implementation for determining the appropriate bucket. Instead of a binary search we can directly determine which bucket a value belongs in with arithmetic, reducing the complexity from `O(log n)` to `O(1)`

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java [hunk 1]
```diff
@@ -95,24 +98,6 @@ private Optional<HistogramBucket> findMatchingBucket(HistogramBucket matchingBuc
     return Optional.empty();
   }
 
-  /**
-   * Runs a binary search to find the correct bucket for the value being inserted and return it's
```
Comment (mansu) [inline kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java:99]: Can we add a unit test comparing the output from this method and the proposed method. That would be a cheap way to make sure that the outputs are identical. 

Also, can we ensure that this method would work in the aggregation path also?

File: kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java [hunk 2]
```diff
@@ -56,8 +59,8 @@ public void add(long value) {
     if (value == high) {
       buckets.get(bucketCount - 1).increment(1);
     } else {
-      int bucketIdx = findMatchingBucket(buckets, 0, bucketCount, (double) value);
-      buckets.get(bucketIdx).increment(1);
+      int index = (int) Math.floor((value - low) / bucketSize);
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java:62]: What happens when value is more than `high`?

Comment (bryanlb) [reply kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java:62]: Good catch! Fixed and added an extra test.

**Review by mansu — CHANGES_REQUESTED** [review]

File: kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java (hunk 2 above)
Comment (vthacker) [reply kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java:64]: FYI I don't think this value will ever be higher than `high` in how we use it :) the collector I think will filter out matching docs in our time range before this gets invoked..

**Comment by vthacker** [comment]
Nice! LGTM

File: kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java (hunk 1 above)
Comment (bryanlb) [reply kaldb/src/main/java/com/slack/kaldb/histogram/FixedIntervalHistogramImpl.java:99]: As discussed via Slack, we verified the new version is generating identical outputs with an uncommitted test prior to removing this old version.

**Review by mansu — APPROVED** [review]

**Pull request merged**